### PR TITLE
HOTFIX: AddressBook requires addresses to have IDs; Do not close conn immediately after sending pex addrs in seed mode

### LINKED ISSUE #2773: p2p: fix logging peer address
Pointed out in https://github.com/tendermint/tendermint/issues/2721#issuecomment-435326581

> module=p2p peer=0xb78f00

It's because we're passing a function pointer rather than calling the function in: https://github.com/tendermint/tendermint/blob/6e9aee546061423864fef3f9fa332ef37c3dca96/p2p/switch.go#L619

We should just pass the NetAddress() there, no need to call String().

But note even if we do that, we still lose the ID. That's because in https://github.com/tendermint/tendermint/blob/6e9aee546061423864fef3f9fa332ef37c3dca96/p2p/netaddress.go#L316-L321 we expect the protocol to be at the beginning, but it could come after the `@`, eg. `76e35576b8dd1cf8049fea382fda8ac621a9c6ee@tcp://0.0.0.0:26656`. So we need fix this too

### PR DESCRIPTION
**Pull request by jaekwon**
This solves an issue of having peers without IDs in the address book.

It also fixes #2773, which was a likely cause of there being peers without IDs in the address book in the first place.

It also includes a temporary hack fix for #2092.

### CODE REVIEW

**Comment by melekes** [comment]
Looks like a lot of commits are from the current `develop`. Was this unintentional?

**Comment by ebuchman** [comment]
Thanks. Retargeted to develop. Will merge and release v0.26.1. 

I took a stab at flushing messages on the conn before stopping in https://github.com/tendermint/tendermint/pull/2802 - should address https://github.com/tendermint/tendermint/pull/2797/files#diff-8c6b1a350a161ced01d68514ccda7484R224

File: p2p/pex/pex_reactor.go [hunk 1]
```diff
@@ -221,6 +221,7 @@ func (r *PEXReactor) Receive(chID byte, src Peer, msgBytes []byte) {
 		// 2) limit the output size
 		if r.config.SeedMode {
 			r.SendAddrs(src, r.book.GetSelectionWithBias(biasToSelectNewPeers))
+			time.Sleep(time.Second * 3) // TODO Rethink this.  Without it, above may not actually send.
```
Comment (ebuchman) [inline p2p/pex/pex_reactor.go]: We should put these in a go-routine so as to not block Receive

File: CHANGELOG.md [hunk 2]
```diff
@@ -139,8 +139,8 @@ increasing attention to backwards compatibility. Thanks for bearing with us!
 - [node] [\#2434](https://github.com/tendermint/tendermint/issues/2434) Make node respond to signal interrupts while sleeping for genesis time
 - [state] [\#2616](https://github.com/tendermint/tendermint/issues/2616) Pass nil to NewValidatorSet() when genesis file's Validators field is nil
 - [p2p] [\#2555](https://github.com/tendermint/tendermint/issues/2555) Fix p2p switch FlushThrottle value (@goolAdapter)
-- [p2p] [\#2668](https://github.com/tendermint/tendermint/issues/2668) Reconnect to originally dialed address (not self-reported
-  address) for persistent peers
+- [p2p] [\#2668](https://github.com/tendermint/tendermint/issues/2668) Reconnect to originally dialed address (not self-reported address) for persistent peers
+- [p2p] [\#2797](https://github.com/tendermint/tendermint/pull/2797) AddressBook requires addresses to have IDs; Do not crap out immediately after sending pex addrs in seed mode
```
Comment (ebuchman) [inline CHANGELOG.md]: Wrong changelog file :o ! Please use CHANGELOG_PENDING.md

**Pull request merged**